mp/src/materialsystem/stdshaders/pbr_dx9.cpp

```diff
@@ -0,0 +1,620 @@
+                }
+            }
+
+            // Enabling sRGB writing
+            // See common_ps_fxc.h line 349
+            // PS2b shaders and up write sRGB
+            pShaderShadow->EnableSRGBWrite(true);
+
+            if (IS_FLAG_SET(MATERIAL_VAR_MODEL))
+            {
+                // We only need the position and surface normal
+                unsigned int flags = VERTEX_POSITION | VERTEX_NORMAL | VERTEX_FORMAT_COMPRESSED;
+                // We need three texcoords, all in the default float2 size
+                pShaderShadow->VertexShaderVertexFormat(flags, 1, 0, 0);
+            }
+            else
+            {
+                // We need the position, surface normal, and vertex compression format
+                unsigned int flags = VERTEX_POSITION | VERTEX_NORMAL;
+                // We only need one texcoord, in the default float2 size
+                pShaderShadow->VertexShaderVertexFormat(flags, 3, 0, 0);
+            }
+        
+            int useParallax = params[info.useParallax]->GetIntValue();
+            if (!mat_pbr_parallaxmap.GetBool()) {
+                useParallax = 0;
+            }
+
+            if (!g_pHardwareConfig->SupportsShaderModel_3_0() || mat_pbr_force_20b.GetBool())
+            {
+
+                // Setting up static vertex shader
+                DECLARE_STATIC_VERTEX_SHADER(pbr_vs20b);
+                SET_STATIC_VERTEX_SHADER(pbr_vs20b);
+
+                // Setting up static pixel shader
+                DECLARE_STATIC_PIXEL_SHADER(pbr_ps20b);
+                SET_STATIC_PIXEL_SHADER_COMBO(FLASHLIGHT, bHasFlashlight);
+                SET_STATIC_PIXEL_SHADER_COMBO(FLASHLIGHTDEPTHFILTERMODE, nShadowFilterMode);
+                SET_STATIC_PIXEL_SHADER_COMBO(LIGHTMAPPED, bLightMapped);
+                SET_STATIC_PIXEL_SHADER_COMBO(EMISSIVE, bHasEmissionTexture);
+                SET_STATIC_PIXEL_SHADER_COMBO(SPECULAR, 0);
+                SET_STATIC_PIXEL_SHADER(pbr_ps20b);
+
+            }
+            else {
+
+                // Setting up static vertex shader
+                DECLARE_STATIC_VERTEX_SHADER(pbr_vs30);
+                SET_STATIC_VERTEX_SHADER(pbr_vs30);
+
+
+                        // Setting up static pixel shader
+                        DECLARE_STATIC_PIXEL_SHADER(pbr_ps30);
+                        SET_STATIC_PIXEL_SHADER_COMBO(FLASHLIGHT, bHasFlashlight);
+                        SET_STATIC_PIXEL_SHADER_COMBO(FLASHLIGHTDEPTHFILTERMODE, nShadowFilterMode);
+                        SET_STATIC_PIXEL_SHADER_COMBO(LIGHTMAPPED, bLightMapped);
+                        SET_STATIC_PIXEL_SHADER_COMBO(USEENVAMBIENT, bUseEnvAmbient);
+                        SET_STATIC_PIXEL_SHADER_COMBO(EMISSIVE, bHasEmissionTexture);
+                        SET_STATIC_PIXEL_SHADER_COMBO(SPECULAR, bHasSpecularTexture);
+                        SET_STATIC_PIXEL_SHADER_COMBO(PARALLAXOCCLUSION, useParallax);
+                        SET_STATIC_PIXEL_SHADER(pbr_ps30);
+
+            }
```

Comment:
Formatting. Fix weird spacing and put curly braces on their own line

Reply:
Will do